packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -417,65 +417,148 @@ contract OPContractsManager is ISemver {
+                // 3. getting the anchor root for the respected game type from the Anchor State Registry.
+                {
+                    GameType gameType = IOptimismPortal2(payable(opChainAddrs.optimismPortal)).respectedGameType();
+                    (Hash root, uint256 l2BlockNumber) = permissionedDisputeGame.anchorStateRegistry().anchors(gameType);
+                    OutputRoot memory startingAnchorRoot = OutputRoot({ root: root, l2BlockNumber: l2BlockNumber });
+
+                    upgradeToAndCall(
+                        _opChains[i].proxyAdmin,
+                        address(newAnchorStateRegistryProxy),
+                        impls.anchorStateRegistryImpl,
+                        abi.encodeCall(
+                            IAnchorStateRegistry.initialize,
+                            (
+                                superchainConfig,
+                                IDisputeGameFactory(opChainAddrs.disputeGameFactory),
+                                IOptimismPortal2(payable(opChainAddrs.optimismPortal)),
+                                startingAnchorRoot
+                            )
+                        )
+                    );
+                }
+
+                // Permissioned Dispute Game Upgrades
+                // 1. Get and upgrade the Permissioned DelayedWETH Proxy
+                // 2. Deploy a new Permissioned Dispute Game
+                // 3. Set the new Permissioned Dispute Game as the implementation on the OptimismPortal
+
+                IDelayedWETH delayedWETHPermissionedGameProxy = permissionedDisputeGame.weth();
+                upgradeTo(_opChains[i].proxyAdmin, address(delayedWETHPermissionedGameProxy), impls.delayedWETHImpl);
+
+                IFaultDisputeGame.GameConstructorParams memory params =
+                    getGameConstructorParams(IFaultDisputeGame(address(permissionedDisputeGame)));
+                params.anchorStateRegistry = IAnchorStateRegistry(address(newAnchorStateRegistryProxy));
+                address proposer = permissionedDisputeGame.proposer();
+                address challenger = permissionedDisputeGame.challenger();
+                IPermissionedDisputeGame newPermissionedDisputeGame = IPermissionedDisputeGame(
+                    Blueprint.deployFrom(
+                        bps.permissionedDisputeGame1,
+                        bps.permissionedDisputeGame2,
+                        computeSalt(l2ChainId, "v2.0.0", "PermissionedDisputeGame"),
+                        encodePermissionedFDGConstructor(params, proposer, challenger)
+                    )
+                );
+                IDisputeGameFactory(opChainAddrs.disputeGameFactory).setImplementation(
+                    GameTypes.PERMISSIONED_CANNON, IDisputeGame(address(newPermissionedDisputeGame))
+                );
+            }
 
             // Now retrieve the permissionless game. If it exists, upgrade its weth and replace its implementation.
```

Comment:
Could/should we encapsulate all this upgrade logic into a helper function that we reuse for both game types? That would help with code size and auditability. Since the calldata args are different we might need that as a function input, or a conditional within the helper method. Could be a future PR too

Reply:
I tried that out, and I think I like it slightly more, WDYT? 082fb21ddcc66c72b4249f4d17ed563f0ec7cdd9